### PR TITLE
Add v23.1.16 release notes

### PR DESCRIPTION
**Pull request by rmloveland**
Fixes DOC-9769

### CODE REVIEW

**Comment by rmloveland** [comment]
hi @taroface ! this one is scheduled for tomorrow (2024-02-27).  Just one commit.  Same fix as v23.2.2 in this PR: https://github.com/cockroachdb/docs/pull/18322

File: src/current/_includes/releases/v23.1/v23.1.16.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+{% include releases/release-downloads-docker-image.md release=include.release %}
+
+<h3 id="v23-1-16-bug-fixes">Bug fixes</h3>
+
+- Fixed a bug where [rangefeed](https://www.cockroachlabs.com/docs/v23.1/use-changefeeds#enable-rangefeeds) resolved timestamps could get stuck, continually emitting the [log message](https://www.cockroachlabs.com/docs/v23.1/logging) `"pushing old intents failed: range barrier failed, range split"`, typically following a [range merge](https://www.cockroachlabs.com/docs/v23.1/architecture/distribution-layer#range-merges). This bug was introduced in v23.1.15. [#119559][#119559]
+
+<h3 id="v23-1-16-contributors">Contributors</h3>
+
+This release includes 2 merged PRs by 2 authors.
```
Comment (taroface) [inline src/current/_includes/releases/v23.1/v23.1.16.md:13]: Is it weird or normal that we name 2 PRs but only have 1 release note?

**Review by taroface — APPROVED** [review]
LGTM

File: src/current/_includes/releases/v23.1/v23.1.16.md (hunk 1 above)
Comment (rmloveland) [reply src/current/_includes/releases/v23.1/v23.1.16.md:13]: not every merged PR is release-note worthy.  IIRC only a subset of total PRs have a release note annotation due to a public-facing change

**Pull request merged**